File: src/tasks_java/NotificationService.java
package tasks_java;

interface Notifier {
    void sendNotification(String message);
}

class NotificationService {
    Notifier notifier;

    NotificationService(Notifier notifier) {
        this.notifier = notifier;
    }

    void notifyUser(String message) {
       notifier.sendNotification(message);
    }
}

class EmailNotifier implements Notifier {
    @Override
    public void sendNotification(String message) {
        System.out.println("Отправка по E-mail: " + message);
    }
}

class SMSNotifier  implements Notifier {
    @Override
    public void sendNotification(String message) {
        System.out.println("Отправка по SMS: " + message);
    }
}

class PushNotifier  implements Notifier {

    @Override
    public void sendNotification(String message) {
        System.out.println("Отправка Push-уведомления: " + message);
    }
}

File: src/tasks_kotlin/NotificationService.kt
package tasks_kotlin

interface Notifier {
    fun sendNotification(message: String)
}
class NotificationService(
    private val notifier: Notifier
) {
    fun notifyUser(message: String) {
        notifier.sendNotification(message)
    }
}

class EmailNotifier: Notifier {
    override fun sendNotification(message: String) {
        println("Отправка по E-mail: $message")
    }
}

class SMSNotifier: Notifier {
    override fun sendNotification(message: String) {
        println("Отправка по SMS: $message")
    }
}

class PushNotifier: Notifier {
    override fun sendNotification(message: String) {
        println("Отправка Push-уведомления: $message")
    }
}